pkg/restapi/vc/operation/operations.go

```diff
@@ -233,24 +322,113 @@ func createMACCrypto() (ariescrypto.Crypto, *keyset.Handle, string, error) {
 		return nil, nil, "", err
 	}
 
-	vcIDIndexNameMAC, err := tinkCrypto.ComputeMAC([]byte(vcIDEDVIndexName), kh)
+	keyHandle, err := prepareKeyHandle(storeProvider, localKMS)
 	if err != nil {
 		return nil, nil, "", err
 	}
 
-	return tinkCrypto, kh, base64.URLEncoding.EncodeToString(vcIDIndexNameMAC), nil
+	vcIDIndexNameMAC, err := tinkCrypto.ComputeMAC([]byte(vcIDEDVIndexName), keyHandle)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	return tinkCrypto, keyHandle, base64.URLEncoding.EncodeToString(vcIDIndexNameMAC), nil
+}
+
+func prepareKeyHandle(storeProvider storage.Provider, localKMS *localkms.LocalKMS) (*keyset.Handle, error) {
+	keyIDStore, err := prepareKeyIDStore(storeProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	var kh *keyset.Handle
+
+	keyIDBytes, err := keyIDStore.Get(keyIDDBKeyName)
+	switch err {
+	case storage.ErrValueNotFound:
+		keyID, keyHandleUntyped, createErr := localKMS.Create(kms.HMACSHA256Tag256Type)
+		if createErr != nil {
+			return nil, createErr
+		}
+
+		var ok bool
+
+		kh, ok = keyHandleUntyped.(*keyset.Handle)
+		if !ok {
+			return nil, errKeySetHandleAssertionFailure
+		}
+
+		err = keyIDStore.Put(keyIDDBKeyName, []byte(keyID))
+		if err != nil {
+			return nil, err
+		}
+	case nil:
+		keyHandleUntyped, getErr := localKMS.Get(string(keyIDBytes))
+		if getErr != nil {
+			return nil, getErr
+		}
+
+		var ok bool
+
+		kh, ok = keyHandleUntyped.(*keyset.Handle)
+		if !ok {
+			return nil, errKeySetHandleAssertionFailure
+		}
+	default:
+		return nil, err
+	}
+
+	return kh, nil
+}
+
+func prepareKeyIDStore(storeProvider storage.Provider) (storage.Store, error) {
+	err := storeProvider.CreateStore(keyIDStoreName)
+	if err != nil {
+		if err != storage.ErrDuplicateStore {
+			return nil, err
+		}
+	}
+
+	return storeProvider.OpenStore(keyIDStoreName)
+}
+
+func prepareMasterKeyReader(kmsSecretsStoreProvider ariesstorage.Provider) (*bytes.Reader, error) {
+	masterKeyStore, err := kmsSecretsStoreProvider.OpenStore(masterKeyStoreName)
+	if err != nil {
+		return nil, err
+	}
+
+	masterKey, err := masterKeyStore.Get(masterKeyDBKeyName)
+	if err != nil {
+		if err == ariesstorage.ErrDataNotFound {
+			masterKeyRaw := random.GetRandomBytes(uint32(32))
+			masterKey = []byte(base64.URLEncoding.EncodeToString(masterKeyRaw))
+
+			putErr := masterKeyStore.Put(masterKeyDBKeyName, masterKey)
+			if putErr != nil {
+				return nil, putErr
+			}
+		} else {
```

Comment:
no need for else

Reply:
If I remove the else, then any errors returned from masterKeyStore.Get(masterKeyDBKeyName) won't be caught.